File: service-template-go/internal/app/controller/response_content.go
package controller

import "service-template-go/internal/app/constant"

type ResponseContent struct {
	Code    int
	Message string
	Data    interface{}
}

func Failed() *ResponseContent {
	return Of(constant.COMMON_BIZ_ERROR.Code, constant.COMMON_BIZ_ERROR.Message, nil)
}

func Success(data interface{}) *ResponseContent {
	return Of(constant.SUCCESS.Code, constant.SUCCESS.Message, data)
}

func Of(code int, message string, data interface{}) *ResponseContent {
	return &ResponseContent{
		code,
		message,
		data,
	}
}

File: service-template-java/src/main/java/app/alphahunter/backend/site/controller/ResponseContent.java
package app.alphahunter.backend.site.controller;

import app.alphahunter.backend.site.constant.BusinessCode;
import lombok.Getter;
import lombok.Setter;

/**
 * 供非rest接口使用, 如AuthenticationController
 *
 * @param <T>
 */
@Getter
@Setter
public class ResponseContent<T> {

    private Integer code;
    private String message;
    private T data;

    private ResponseContent() {

    }

    public static <T> ResponseContent<T> success() {
        return success(null);
    }

    public static <T> ResponseContent<T> success(T data) {
        ResponseContent<T> responseContent = new ResponseContent<>();
        responseContent.setCode(BusinessCode.SUCCESS.getCode());
        responseContent.setMessage("");
        responseContent.setData(data);
        return responseContent;
    }

    public static <T> ResponseContent<T> of(Integer code, String message, T data) {
        ResponseContent<T> responseContent = new ResponseContent<>();
        responseContent.setCode(code);
        responseContent.setMessage(message);
        responseContent.setData(data);
        return responseContent;
    }
}
